src/dgipy/dgidb.py

```diff
@@ -29,21 +28,29 @@ def _get_client(api_url: str) -> Client:
     return Client(transport=transport, fetch_schema_from_transport=True)
 
 
+def _group_attributes(row: list[dict]) -> dict:
+    grouped_dict = {}
+    for attr in row:
+        if attr["name"] in grouped_dict:
+            grouped_dict[attr["name"]].append(attr["value"])
+        else:
+            grouped_dict[attr["name"]] = [attr["value"]]
+    return grouped_dict
```

Comment:
testing this part around a little bit. Will this (+the code from get_interactions) basically see all possible keys from a search result and make them as fields within this column for all entries? I noticed right away playing around with the new implementation a lot of nulls everywhere:

![Screenshot 2024-08-29 at 8 37 15 AM](https://github.com/user-attachments/assets/12769b5f-2ea5-4a72-90cf-1f39cf2091a3)

I think this is fine and I think I like this implementation better but it makes me wonder: 1) will users be confused if they see nulls (i.e. will they think something broke) and 2) will people be confused if search to search the fields pop in and out (for example what if Cancer Type appears for one search and suddenly the whole column of one particular search result has Cancer Type: None for 95% of entries). These could also not really be concerns but just wanted to think about them. 

Related, all the types of interaction attributes in the db right now:
![image](https://github.com/user-attachments/assets/436e9632-4909-4582-a90f-6fcdab2961d7)

Reply:
example of "pop-in" from get_drug. One of these entries has a "Developer" data point that feels fairly unique, and all other entries have null there now:

![Screenshot 2024-08-29 at 8 53 02 AM](https://github.com/user-attachments/assets/d69c562f-c25c-4092-ae97-75f8e81eb5c4)